### PR TITLE
Fix `distributed_concat` with scalar tensor

### PR DESCRIPTION
**Pull request by Yard1**
# What does this PR do?

If a scalar tensor is passed to `distributed_concat`, the output tensors are correctly converted to one element vectors. However, this is not done for the tensor itself, which causes an exception to be thrown in `dist.all_gather` due to a tensor length mismatch. This PR fixes that.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Yard1** [comment]
cc @sgugger

File: src/transformers/trainer_pt_utils.py [hunk 1]
```diff
@@ -161,6 +161,8 @@ def distributed_concat(tensor: Any, num_total_examples: Optional[int] = None) ->
             return type(tensor)(distributed_concat(t, num_total_examples) for t in tensor)
         output_tensors = [tensor.clone() for _ in range(dist.get_world_size())]
         output_tensors = [t if len(t.shape) > 0 else t[None] for t in output_tensors]
+        if len(tensor.shape) <= 0:
+            tensor = tensor[None]
```
Comment (sgugger) [inline src/transformers/trainer_pt_utils.py:165]: Maybe we should move this before the definition of `output_tensors` so we can remove current line 163?

**Review by sgugger — COMMENTED** [review]
Thanks for fixing this! Just left some suggestion to make the code simpler.

File: src/transformers/trainer_pt_utils.py (hunk 1 above)
Comment (Yard1) [reply src/transformers/trainer_pt_utils.py:163]: Done!

**Review by sgugger — APPROVED** [review]

**Pull request merged**